tools/autograd/templates/VariableType.cpp

```diff
@@ -446,6 +446,46 @@ Tensor VariableType::contiguous(const Tensor & self) const {
   return self.clone();
 }
 
+Tensor VariableType::detach(const Tensor & self) const {
```

Comment:
What happened here?

Reply:
I'm not sure exactly what you mean.  I copied the generated VariableType::detach() / VariableType::detach_() code, but replaced the dispatch to the TypeDefault with the Variable dispatch call.

Reply:
OK, so I guess your point is the codegen didn't know how to actually do the `NON_GENERATED_CODE` section. It's a bit annoying that we have to do it this way, esp. because there are going to be more methods that will get this treatment, right?